src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -13,6 +13,12 @@ public bool Contains(string value)
             if (value == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
 
+            if (RuntimeHelpers.IsKnownConstant(value) && value.Length == 1)
```

Comment:
If IsKnownConstant(value) is true, presumably the JIT is also already eliding the above null check?

Reply:
> If IsKnownConstant(value) is true, presumably the JIT is also already eliding the above null check?

Yep!